### PR TITLE
Drop Redux

### PR DESCRIPTION
**Pull request by nikitayutanov**
Remove Redux, add global state context providers.

### CODE REVIEW

File: website/frontend/src/components/App/App.tsx [hunk 1]
```diff
@@ -54,34 +55,25 @@ const utilRoutes = [routes.privacyPolicy, routes.termsOfUse];
...
+  const alert = useAlert();
+  const { isLoading } = useLoading();
   const [searchParams, setSearchParams] = useSearchParams();
-  const { isApiReady } = useSelector((state: RootState) => state.api);
-  const { isProgramUploading, isMessageSending } = useSelector((state: RootState) => state.programs);
   const events = useEvents();
 
   useEffect(() => {
-    if ((isProgramUploading || isMessageSending) && document.body.style.overflowY !== 'hidden') {
+    if (isLoading && document.body.style.overflowY !== 'hidden') {
```
Comment (codev0) [inline website/frontend/src/components/App/App.tsx:65]: could you please leave here TODO about change this hack to useBodyScrollLock hook?

File: website/frontend/src/components/pages/Message/Message.tsx [hunk 2]
```diff
@@ -1,45 +1,37 @@
...
-    };
-  }, [dispatch, messageId]);
+    getMessage(messageId).then(({ result }) => setMessage(result));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   useEffect(() => {
     if (message) {
-      dispatch(getProgramAction(message.source));
+      getProgram(message.source).then(({ result }) => setProgram(result));
```
Comment (codev0) [inline website/frontend/src/components/pages/Message/Message.tsx:32]: we should think about add error catching in cases like that

File: website/frontend/src/context/editor/Provider.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import { useAlert } from 'react-alert';
+import JSZip from 'jszip';
+import saveAs from 'file-saver';
+import { EditorContext } from './Context';
+import { Props } from '../types';
+import { LOCAL_STORAGE, WASM_COMPILER_GET } from 'consts';
+
+const { Provider } = EditorContext;
+
+const useEditor = () => {
```
Comment (codev0) [inline website/frontend/src/context/editor/Provider.tsx:11]: this code looks like method or separate hook but not context i think

**Review by codev0 — APPROVED** [review]

**Review by aderrod94 — APPROVED** [review]

**Review by EugenWay — APPROVED** [review]

File: website/frontend/src/components/App/App.tsx (hunk 1 above)
Comment (nikitayutanov) [reply website/frontend/src/components/App/App.tsx:65]: It's obvious that this part should be refactored so no need for additional comment I guess

File: website/frontend/src/components/BlocksSummary/BlocksSummary.tsx [hunk 4]
```diff
@@ -1,14 +1,10 @@
...
-import { RootState } from 'store/reducers';
+import { useApi, useBlocks } from 'hooks';
 import styles from './BlocksSummary.module.scss';
 
 const BlocksSummary = () => {
-  const [api] = useApi();
-  const dispatch = useDispatch();
-
-  const { blocks } = useSelector((state: RootState) => state.blocks);
+  const { api } = useApi();
```
[the remark this answers is not in the document]
Comment (nikitayutanov) [reply website/frontend/src/components/BlocksSummary/BlocksSummary.tsx:6]: Maybe, gonna think about it.

File: website/frontend/src/components/pages/Message/Message.tsx (hunk 2 above)
Comment (nikitayutanov) [reply website/frontend/src/components/pages/Message/Message.tsx:32]: Yeah, we definitely should. Created a separate task for this.

File: website/frontend/src/context/editor/Provider.tsx (hunk 3 above)
Comment (nikitayutanov) [reply website/frontend/src/context/editor/Provider.tsx:11]: Previously it was in the `Header` component. I agree, it's overall massive and awkward, hence worth refactoring.

**Pull request merged**